main/src/propagator/ipropagator.hpp

```diff
@@ -34,7 +34,10 @@
 
 #include <variant>
 
+#include "cstone/sfc/box.hpp"
+#include "cstone/tree/accel_switch.hpp"
 #include "io/ifile_io.hpp"
+#include "sph/particles_data.hpp" //TODO: why we didn't need it before? Should we use cpu or gpu version?
```

Comment:
The additional particles_data header  is needed because of the call to the transferToHost function in propagator.hpp (or ipropagator.hpp as it was called in this PR) and similarly for cstone/sfc/box.hpp and cstone/tree/accel_switch.hpp. In the original propagator.hpp file, according to gcc include chain (-H), the particles_data.hpp header is included anyway thanks to other inclusions, but for some reason that recursive inclusion doesn't work in this case.